app.js

```diff
@@ -136,17 +135,16 @@ if (Config.crashguard) {
 
 global.Sockets = require('./sockets');
 
-exports.listen = function (port, bindAddress, workerCount) {
-	Sockets.listen(port, bindAddress, workerCount);
+exports.listen = function (port) {
```

Comment:
It would be the only block reading `process.argv`. Something roughly like this was what I was thinking of:

```
if (require.main === module) {
	// Launch the server directly when app.js is the main module. Otherwise,
	// in the case of app.js being imported as a module (e.g. unit tests),
	// postpone launching until app.listen() is called.
	let port;
	let bindAddress;
	let workerCount;
	if (process.argv[2]) {
		port = parseInt(process.argv[2]);
		if (port < 1 || port > 65535) port = Config.port;
	}
	if (process.argv[3]) {
		bindAddress = process.argv[3].trim();
		if (bindAddress.toLowerCase() === 'localhost') {
			bindAddress = 'localhost';
		} else {
			let parts = bindAddress.split('.').map(i => parseInt(i));
			if (parts.length !== 4 || !parts.every(i => !isNaN(i) && i >= 0 && i <= 255)) {
				bindAddress = Config.bindaddress;
			}
		}
	}
	if (process.argv[4]) {
		workerCount = parseInt(process.argv[4]);
		if (isNaN(workerCount) || workerCount < 1) workerCount = Config.workers;
	}

	Sockets.listen(port, bindAddress, workerCount);
}
```